sim-lib/src/lib.rs

```diff
@@ -133,6 +133,14 @@ pub enum SimulationError {
     FileError,
     #[error("{0}")]
     RandomActivityError(RandomActivityError),
+    #[error("System Time Error: {0}")]
+    SystemTimeError(#[from] SystemTimeError),
+    #[error("Missing Node Error: {0}")]
+    MissingNodeError(String),
+    #[error("Mpsc Channel Error: {msg}: {error_str}")]
+    MpscChannelError { msg: String, error_str: String },
```

Comment:
I've seen this pattern around rust codebase quite a bit, could you explain the rationale for two fields here when we're just logging both values? Seems like we could just have `MpscChannelError(String)`, though totally open to me being completely ignorant about rust conventions here - shan't be the last time 🦀

Reply:
Please ignore this. I have updated the variant to `MpscChannelError(String)`. 
I use this style when I want to add some context to an internal error with multiple variants, and control logic based on these variants. It has no use here (possible did this out of habit)